# Case Study on Testing of Hypothesis

### A company started to invest in digital marketing as a new way of their product promotions. For that they collected data and decided to carry out a study on it.

### ● The company wishes to clarify whether there is any increase in sales after stepping into digital marketing.
### ● The company needs to check whether there is any dependency between the features “Region” and “Manager”.

### Help the company to carry out their study with the help of data provided.

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import scipy.stats as stats
from scipy.stats import chi2_contingency 

In [2]:
df=pd.read_csv('Desktop\ICTK\Case Study #03 - Testing of Hypothesis\Sales_add.csv')
df

Month      Region      Manager  Sales_before_digital_add(in $)  \
0    Month-1  Region - A  Manager - A                          132921   
1    Month-2  Region - A  Manager - C                          149559   
2    Month-3  Region - B  Manager - A                          146278   
3    Month-4  Region - B  Manager - B                          152167   
4    Month-5  Region - C  Manager - B                          159525   
5    Month-6  Region - A  Manager - B                          137163   
6    Month-7  Region - C  Manager - C                          130625   
7    Month-8  Region - A  Manager - A                          131140   
8    Month-9  Region - B  Manager - C                          171259   
9   Month-10  Region - C  Manager - B                          141956   
10  Month-11  Region - B  Manager - C                          159339   
11  Month-12  Region - A  Manager - A                          178939   
12  Month-13  Region - B  Manager - A                          145062   
13  Month-14  Region - A  Manager - A                          151514   
14  Month-15  Region - C  Manager - A                          147463   
15  Month-16  Region - A  Manager - B                          177195   
16  Month-17  Region - A  Manager - C                          140862   
17  Month-18  Region - C  Manager - B                          167996   
18  Month-19  Region - B  Manager - A                          132135   
19  Month-20  Region - A  Manager - B                          152493   
20  Month-21  Region - B  Manager - A                          147425   
21  Month-22  Region - A  Manager - C                          130263   

    Sales_After_digital_add(in $)  
0                          270390  
1                          223334  
2                          244243  
3                          231808  
4                          258402  
5                          256948  
6                          222106  
7                          230637  
8                          226261  
9                          193735  
10                         203175  
11                         276279  
12                         205180  
13                         253131  
14                         229336  
15                         187305  
16                         234908  
17                         191517  
18                         227040  
19                         212579  
20                         263388  
21                         243020

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22 entries, 0 to 21
Data columns (total 5 columns):
 #   Column                          Non-Null Count  Dtype 
---  ------                          --------------  ----- 
 0   Month                           22 non-null     object
 1   Region                          22 non-null     object
 2   Manager                         22 non-null     object
 3   Sales_before_digital_add(in $)  22 non-null     int64 
 4   Sales_After_digital_add(in $)   22 non-null     int64 
dtypes: int64(2), object(3)
memory usage: 1008.0+ bytes


In [4]:
df.describe()

Sales_before_digital_add(in $)  Sales_After_digital_add(in $)
count                       22.000000                      22.000000
mean                    149239.954545                  231123.727273
std                      14844.042921                   25556.777061
min                     130263.000000                  187305.000000
25%                     138087.750000                  214960.750000
50%                     147444.000000                  229986.500000
75%                     157627.500000                  250909.000000
max                     178939.000000                  276279.000000

## Hypothesis testing; is there an increase in sales after stepping into digital marketing?

#### One tailed two sample t-test

In [5]:
sales_before=df['Sales_before_digital_add(in $)']
sales_after=df['Sales_After_digital_add(in $)']

tvalue, pvalue=stats.ttest_rel(sales_after,sales_before,alternative='greater')
print("The t value of means is: ", round(tvalue,6))
print("The p value based on the t score is: ", round(pvalue,6))

stats.ttest_rel(sales_after,sales_before,alternative='greater')

The t value of means is:  12.090705
The p value based on the t score is:  0.0


Ttest_relResult(statistic=12.09070525287017, pvalue=3.168333502287889e-11)

#### Inferences

###### This hypothesis test gives a p-value of 0 & t-value or t-statistics is 12.0907. Since the p-value simply tells the strength of evidence in support of a null hypothesis, less than the significance level, we reject the null hypothesis & fail to reject the alternate hypothesis.
###### Conclusion: Sales after digital marketing is higher than before digital marketing.

##  The company needs to check whether there is any dependency between the features “Region” and “Manager”.

#### Chi-Square test for Independence

In [6]:
df3=pd.crosstab(df.Region, df.Manager, margins=True)
df3

Manager     Manager - A  Manager - B  Manager - C  All
Region                                                
Region - A            4            3            3   10
Region - B            4            1            2    7
Region - C            1            3            1    5
All                   9            7            6   22

In [7]:
value = np.array([df3.iloc[0][0:3].values,
                  df3.iloc[1][0:3].values,
                  df3.iloc[2][0:3].values])
stat, p, dof, expected=chi2_contingency(value)

print(chi2_contingency(value)[0:3])
alpha = 0.05
print("p value is " + str(p))
if p <= alpha:
    print('Dependent (reject H0)')
else:
    print('Independent (H0 holds true)')

(3.050566893424036, 0.5493991051158094, 4)
p value is 0.5493991051158094
Independent (H0 holds true)


####  Inferences

###### p-vale 0.549399 > alpha of .05
###### Since the p-value is greater than alpha, fails to reject the null hypothesis (H0) and reject the alternate hypothesis (Ha).
###### Conclusion: “Region” and “Manager” in the data are independent and there are no dependency of these features.